python/ray/_raylet.pyx

```diff
@@ -452,14 +455,22 @@ cdef execute_task(
                 actor_title = "{}({}, {})".format(
                     class_name, repr(args), repr(kwargs))
                 core_worker.set_actor_title(actor_title.encode("utf-8"))
+            # Ensure no previous signals are still around
+            check_signals()
```

Comment:
What signals is this checking for? If it's the keyboardinterrupt from interrupt_main, don't we need to catch that and handle it?

Reply:
It was initially just to 'clear' any signals, but I can make it also handle cancellation.